Controllers/ApplicationController.cs

```diff
@@ -150,14 +152,14 @@ public Task<string> FetchBalances()
         [HttpGet]
         public Task<string> FetchBalancesPopup()
         {
-            var queryParams = Request.QueryString;
+            var queryParams = Request.Url.AbsoluteUri;
 
             // retrieve CSRF token from browser cookie
             var csrfToken = Request.Cookies["csrf_token"];
 
             return GetPfmMember() 
                 // check CSRF token and retrieve state and token ID from callback parameters
-                .FlatMap(mem => tokenClient.ParseTokenRequestCallbackParams(queryParams, csrfToken.Value)
+                .FlatMap(mem => tokenClient.ParseTokenRequestCallbackUrl(queryParams, csrfToken.Value)
```

Comment:
Why the change to ParseTokenRequestCallbackUrl?

Reply:
ParseTokenRequestCallbackParams was missing in Tpp SDK. We have added that to Tpp SDK and created the PR: https://github.com/tokenio/sdk-csharp/pull/58 . Can you please review it.
I also reverted the change here as well to use ParseTokenRequestCallbackParams.